traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/ds/DeliveryService.java

```diff
@@ -658,4 +682,61 @@ public boolean isAcceptHttp() {
 	public void setConsistentHashRegex(final String consistentHashRegex) {
 		this.consistentHashRegex = consistentHashRegex;
 	}
+
+	/**
+	 * Extracts the significant parts of a request's query string based on this
+	 * Delivery Service's Consistent Hashing Query Parameters
+	 * @param r The request from which to extract query parameters
+	 * @return The parts of the request's query string relevant to consistent
+	 *	hashing. The result is URI-decoded - if decoding fails it will return
+	 *	a blank string instead.
+	 */
+	public String extractSignificantQueryParams(HTTPRequest r) {
+		if (r.getQueryString() == null || r.getQueryString().isEmpty()) {
```

Comment:
We should probably also add `|| this.getConsistentHashQueryParams().isEmpty()` to this conditional -- if it's empty the result should always be `""`.